src/ioctls/vcpu.rs

```diff
@@ -417,6 +422,144 @@ mod tests {
         assert!(vm.create_vcpu(0).is_ok());
     }
 
+    #[cfg(target_arch = "x86_64")]
+    #[test]
+    fn test_set_cpuid2() {
+        let kvm = Kvm::new().unwrap();
+        if kvm.check_extension(Cap::ExtCpuid) {
+            let vm = kvm.create_vm().unwrap();
+            let mut cpuid = kvm.get_supported_cpuid(MAX_KVM_CPUID_ENTRIES).unwrap();
+            assert!(cpuid.mut_entries_slice().len() <= MAX_KVM_CPUID_ENTRIES);
+            let nr_vcpus = kvm.get_nr_vcpus();
+            for cpu_id in 0..nr_vcpus {
+                let vcpu = vm.create_vcpu(cpu_id as u8).unwrap();
+                vcpu.set_cpuid2(&cpuid).unwrap();
+            }
+        }
+    }
+
+    #[cfg(target_arch = "x86_64")]
+    #[allow(non_snake_case)]
+    #[test]
+    fn test_fpu() {
+        // as per https://github.com/torvalds/linux/blob/master/arch/x86/include/asm/fpu/internal.h
+        let KVM_FPU_CWD: usize = 0x37f;
+        let KVM_FPU_MXCSR: usize = 0x1f80;
+        let kvm = Kvm::new().unwrap();
+        let vm = kvm.create_vm().unwrap();
+        let vcpu = vm.create_vcpu(0).unwrap();
+        let mut fpu: kvm_fpu = kvm_fpu {
+            fcw: KVM_FPU_CWD as u16,
+            mxcsr: KVM_FPU_MXCSR as u32,
+            ..Default::default()
+        };
+
+        fpu.fcw = KVM_FPU_CWD as u16;
+        fpu.mxcsr = KVM_FPU_MXCSR as u32;
+
+        vcpu.set_fpu(&fpu).unwrap();
+        assert_eq!(vcpu.get_fpu().unwrap().fcw, KVM_FPU_CWD as u16);
+    }
+
+    #[cfg(target_arch = "x86_64")]
+    #[test]
+    fn lapic_test() {
+        use std::io::Cursor;
+        //we might get read of byteorder if we replace 5h3 mem::transmute with something safer
```

Comment:
What is `5h3`?

Reply:
3 characters from one of my passwords, but I won't tell you which one! I'll remove it.